pptx_blueprint/__init__.py

```diff
@@ -87,6 +89,21 @@ def _find_shapes_in_slide(slide):
 
         return matched_shapes
 
+    def _get_all_shapes(self) -> Iterable[BaseShape]:
+        # Do we need all the shapes? Perhaps we should filter on tags here.
+        all_shapes = [shape for slide in self._presentation.slides for shape in slide.shapes]
+        return all_shapes
+
+    def _copy_tags_to_name(self) -> None:
+        all_shapes = self._get_all_shapes()
+        # This regex matches on tags
+        regex_tag = re.compile(r'^\s*(\{\w+\})\s*$')
```

Comment:
Could this be defined as a class constant?

Reply:
Further, why are the curly brackets part of the group? Do we really want to set the shape.name to e.g. '{location}' instead of 'location'?

Reply:
> Further, why are the curly brackets part of the group? Do we really want to set the shape.name to e.g. '{location}' instead of 'location'?

Tim and I had discussed this, yes. Leaving the curly braces in makes it clear that this is a tag.

Reply:
> Could this be defined as a class constant?

 Could do, but if so, please private. I don't want to make this user-configurable as of now. Strictly speaking, it would even be sufficient not to explicitly compile the regexp and just use inline `if re.match(r'^\s*(\{\w+\})\s*$', shape.text):`. `re` caches the last few (I belive to remember 100?) expressions. Therefore, the overhead for this basic call vs. the compiled regexp is just a dict lookup, which is negligible.

> Do we really want to set the shape.name to e.g. '{location}' instead of 'location'?
I tend to say yes. While the curly brace feels a bit bulky here, it serves to highly reduce the likelihood of a name collision between a tag and an existing shape name.